### DISCUSSION TITLE
Is it expected for gocron to execute function as soon as the scheduler is started?

### DISCUSSION THREAD

**Discussion by CodeEighty8** [go-co-op/gocron#407] [Q&A] [answered]
I am trying out gocron and noticed that although the task is scheduled X seconds in the future, it seems to get executed immediately after the scheduler is started. Is this an expected behavior? 

There seems to be a similar issue reported earlier as well and fixed but I am still noticing it. 
https://github.com/go-co-op/gocron/issues/60
In fact, the following example is picked up from one of the comments where the author said, it's not reproducible.
(https://github.com/go-co-op/gocron/issues/60#issuecomment-692787370)

In the attached screenshot, the scheduler gets started with this logline: `Starting.. 2009-11-10 23:00:00 +0000 UTC m=+0.000000001` and immediately the first run happens at the same time with `DONE 2009-11-10 23:00:00 +0000 UTC m=+0.000000001`
And 

Go Playground link: https://go.dev/play/p/Wv9lQJF8Wjj

<img width="543" alt="image" src="https://user-images.githubusercontent.com/7677904/209173978-2e5091bb-c319-40f6-9278-2ea46ec00a58.png">

**Comment by JohnRoesler** [comment, marked as answer]
https://github.com/go-co-op/gocron/discussions/201 this question is answered here

**Reply by CodeEighty8** [reply]
Thank you very much for pointing me in the right direction. 🙏
